test/unittests/validation_test.cpp

```diff
@@ -52,6 +52,19 @@ TEST(validation, globals_invalid_initializer)
     const auto bin_invalid3 = from_hex("0061736d01000000060b027f00412a0b7f0123000b");
     EXPECT_THROW_MESSAGE(parse(bin_invalid3), validation_error,
         "global can be initialized by another const global only if it's imported");
+
+    /* wat2wasm --no-check
+      (global (mut i64) (i32.const 16) (i64.const -1))
```

Comment:
```suggestion
      (global (mut i64) (i64.const 16) (i64.const -1))
```
To make sure it fails on instructions count, not individual type mismatch.